### PR TITLE
Download and install dependencies for WDA

### PR DESCRIPTION
**Pull request by rafael-chavez**
new web driver requires installing dependencies via carthage. Cloning out of the box doesn't have dependencies. createXcodeBuildSubProcess will fail.

### CODE REVIEW

**Comment by rafael-chavez** [comment]
@jlipps @imurchie one small addition, and small fixes.

File: lib/commands/gesture.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ helpers.mobileScroll = async function (opts={}) {
   // proxy scroll request
   let body = opts;
   let method = 'POST';
-  let endpoint = '/uiaElement' + opts.element + '/scroll';
+  let endpoint = '/uiaElement/' + opts.element + '/scroll';
```
Comment (jlipps) [inline lib/commands/gesture.js]: actually can we make this a template string?

```js
let endpoint = `/uiaElement/${opts.element}/scroll`;
```

so make it more clear?

File: lib/webdriveragent.js [hunk 2]
```diff
@@ -108,6 +110,18 @@ class WebDriverAgent {
     });
   }
 
+  async checkForDependencies() {
+    if (await fs.hasAccess(this.bootstrapPath + "/Carthage")) {
+      this.xcodebuild.start();
+      return;
+    }
+    let proc = new SubProcess('/bin/bash', ['Scripts/bootstrap.sh'], {cwd: this.bootstrapPath});
```
Comment (jlipps) [inline lib/webdriveragent.js]: I would say here you would just use `teen_process.exec`:

```js
await exec('/bin/bash', ['Scripts/bootstrap.sh'], {cwd: this.bootstrapPath});
this.xcodebuild.start();
```

another question, what is `xcodebuild.start()` and is it not async?

File: lib/commands/gesture.js (hunk 1 above)
Comment (rafael-chavez) [inline lib/commands/gesture.js]: Sure thing. I'll update.

File: lib/webdriveragent.js (hunk 2 above)
Comment (rafael-chavez) [inline lib/webdriveragent.js]: oh this.xcodebuild is set in line 68 and its just a SubProcess (which wraps teen_process spawn).

```
this.xcodebuild = this.createXcodeBuildSubProcess();
```

start() method is async, but we didnt wait for it previously (line 79)

Comment (jlipps) [inline lib/webdriveragent.js]: I still think we shouldn't be using `SubProcess` to call `bootstrap.sh`.

And probably not awaiting `xcodebuild.start()` is a bug.

Comment (rafael-chavez) [inline lib/webdriveragent.js]: ill update this to fix these two issues.

**Comment by rafael-chavez** [comment]
updated @jlipps

**Comment by jlipps** [comment]
@rafael-chavez LGTM. did you run it with this new code and observe it working? looks like tests are down ATM

**Comment by rafael-chavez** [comment]
@jlipps these are the scenarios i tested:

1. Run on fresh clone ( git clone --recursive https://github.com/appium/appium-xcuitest-driver.git). Here the dependencies are missing (Carthage folder is not there).
2. After first run (dependencies get downloaded and webdriver is built), I re-run. The driver is successfully started and the dependency process does not get called.
3. After several runs, i deleted the webdriver from the root repository (appium-xcuitest-driver) and i re-cloned the WebDriver agent, and ran the driver. The depedencies were re-downloaded and the driver runs successfully. 

Technically 3 should never happen, but I tested it to make sure the behavior is correct.

**Comment by jlipps** [comment]
@rafael-chavez rockin \o/

**Comment by jlipps** [comment]
published as 1.0.4

**Pull request merged**